src/env.rs

```diff
@@ -47,13 +47,23 @@ pub fn get_remaining_points_for_instance(instance: &Instance) -> u64 {
     }
 }
 
-pub fn sub_remaining_point(env: &Env, points: u64) {
+pub fn sub_remaining_point(env: &Env, points: u64) -> ABIResult<()> {
     let instance = &env.instance.clone().unwrap();
     let remaining_points = get_remaining_points_for_env(env);
     if let Some(remaining_points) = remaining_points.checked_sub(points) {
         set_remaining_points(instance, remaining_points);
     } else {
-        abi_bail!("Remaining point reach zero");
+        abi_bail!("Remaining point reach zero")
+    }
+    Ok(())
+}
+
+/// Try to substract remaining point computing the points with a*b and ceiling
+/// the result.
+pub fn sub_remaining_points_with_mult(env: &Env, a: usize, b: usize) -> ABIResult<()> {
+    match a.checked_mul(b) {
+        Some(points) => sub_remaining_point(env, points as u64),
+        None => abi_bail!("Failed to substract points"),
```

Comment:
overflow when multiplying gas by gas price (it's not a failre of the substraction)

Reply:
typo fixed